test/FunctionalTests/Microsoft.OData.Core.Tests/Json/ODataUtf8JsonWriterTests.cs

```diff
@@ -169,6 +169,24 @@ public void WritePrimitiveValueString()
             this.VerifyWritePrimitiveValue("string", "\"string\"");
         }
 
+        [Fact]
+        public void WritePrimitiveValueLargeString()
+        {
+            this.VerifyWritePrimitiveValue(
+                new string('x', 30000),
+                "\"" + new string('x', 30000) + "\""
+                );
+        }
+
+        [Fact]
+        public void WritePrimitiveValueLargeStringWithSpecialChars()
+        {
+            this.VerifyWritePrimitiveValue(
+                new string('x', 20000) + "Foo ия" + new string('x', 10000),
+                "\"" + new string('x', 20000) + "Foo \\u0438\\u044F" + new string('x', 10000) + "\""
+                );
+        }
+

```

Comment:
I think you should add tests with strings that contain surrogate pairs at chunk boundaries. E.g. when one half of the surrogate pair is at the end of one chunk, and the other half is at the beginning of the other half.